make doctest-cli fails on CPP #if

I'm documenting some makefile targets for #10596. I tried the `doctest-cli` target but it fails.

https://github.com/haskell/cabal/blob/4da153d218e588d819d18020f29714e800197326/Makefile#L134-L137

```
$ make doctest-cli
doctest -D__DOCTEST__ --fast cabal-install/src cabal-install-solver/src cabal-install-solver/src-assertion

cabal-install/src/Distribution/Client/HashValue.hs:74:0: error:
     error: missing binary operator before token "("
       74 | #if MIN_VERSION_base16_bytestring(1,0,0)
   |
74 | #if MIN_VERSION_base16_bytestring(1,0,0)
   | ^

cabal-install/src/Distribution/Client/HttpUtils.hs:244:0: error:
     error: missing binary operator before token "("
      244 | #if MIN_VERSION_base16_bytestring(1,0,0)
    |
244 | #if MIN_VERSION_base16_bytestring(1,0,0)
    | ^

cabal-install/src/Distribution/Client/Compat/Tar.hs:7:0: error:
     error: missing binary operator before token "("
        7 | #if MIN_VERSION_tar(0,6,0)
  |
7 | #if MIN_VERSION_tar(0,6,0)
  | ^

cabal-install/src/Distribution/Client/Compat/Tar.hs:26:0: error:
     error: missing binary operator before token "("
       26 | #if MIN_VERSION_tar(0,6,0)
   |
26 | #if MIN_VERSION_tar(0,6,0)
   | ^

cabal-install/src/Distribution/Client/Compat/Tar.hs:54:0: error:
     error: missing binary operator before token "("
       54 | #if MIN_VERSION_tar(0,6,0)
   |
54 | #if MIN_VERSION_tar(0,6,0)
   | ^

cabal-install/src/Distribution/Client/Compat/Tar.hs:1:1: error:
    `gcc' failed in phase `C pre-processor'. (Exit code: 1)
  |
1 | {-# LANGUAGE CPP #-}
  | ^

cabal-install/src/Distribution/Client/HashValue.hs:1:1: error:
    `gcc' failed in phase `C pre-processor'. (Exit code: 1)
  |
1 | {-# LANGUAGE CPP #-}
  | ^

cabal-install/src/Distribution/Client/HttpUtils.hs:1:1: error:
    `gcc' failed in phase `C pre-processor'. (Exit code: 1)
  |
1 | {-# LANGUAGE BangPatterns #-}
  | ^
make: *** [Makefile:137: doctest-cli] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -157,11 +157,6 @@ doctest: ## Run doctests.
 	cd cabal-install-solver && $(DOCTEST)
 	cd cabal-install && $(DOCTEST)
 
-# This is not run as part of validate.sh (we need hackage-security, which is tricky to get).
-.PHONY: doctest-cli
-doctest-cli :
-	doctest -D__DOCTEST__ --fast cabal-install/src cabal-install-solver/src cabal-install-solver/src-assertion
-
 .PHONY: doctest-install
 doctest-install: ## Install doctest tool needed for running doctests.
 	cabal install doctest --overwrite-policy=always --ignore-project --flag cabal-doctest

